blockprotocol/blockprotocol
# Configure lockfile-lint

Context:
- https://snyk.io/blog/why-npm-lockfiles-can-be-a-security-blindspot-for-injecting-malicious-modules/
- https://dev.to/kozlovzxc/injecting-backdoors-to-npm-packages-a0k
- https://www.npmjs.com/package/lockfile-lint
- [Slack thread](https://hashintel.slack.com/archives/C02TWBTT3ED/p1645521936311409) (internal)
- [Asana task](https://app.asana.com/0/0/1201864747725854) (internal)

If you want to test this change locally:

1. Run `yarn install`

1. Run `yarn lint:lockfile-lint` and observe

   ```
   ✔ No issues detected
   ```

1. Open `yarn.lock`

1.  Replace any mention of `https://registry.yarnpkg.com/...` with `https://registry.example.com/...`

1. Run `yarn lint:lockfile-lint` and observe

    ```
    detected invalid host(s) for package: PACKAGE_NAME
    expected: registry.yarnpkg.com
    actual: registry.example.com

   ✖ Error: security issues detected! 
    ```

See also: https://github.com/hashintel/hash/pull/386
